src/controllers/redeploy.go

```diff
@@ -1,20 +1,45 @@
 package controllers
 
 import (
+	"errors"
+	"fmt"
+
 	"github.com/devclub-iitd/DeployBot/src/history"
+	"github.com/devclub-iitd/DeployBot/src/slack"
+	log "github.com/sirupsen/logrus"
 )
 
 func redeploy(callbackID string, data map[string]interface{}) {
+	log.Infof("service(%s) is being stopped", data["git_repo"].(string))
 	stop(callbackID, data)
+	channel := data["channel"].(string)
 	state := history.GetState(data["git_repo"].(string))
 	if state.Status == "stopped" {
-		if _, ok := data["subdomain"]; !ok && state.Subdomain != "" {
+		if _, ok := data["subdomain"]; !ok {
+			if state.Subdomain == "" {
+				log.Infof("Subdomain Has Not Been Set")
+				var err = errors.New("Subdomain Has Not Been Set")
```

Comment:
Don't log the same thing as the error. Log should have more context. Do something like
```
err := errors.New("Subdomain not set")
log.Errorf("Could not deploy service - %v", err)
```
Do the same for the other two

Reply:
Please DON'T  use nested ifs. You can use && (boolean and)